### PR TITLE
Add “Action” Tweaks (from PR #90)

### PR DESCRIPTION
**Pull request by bryanjclark**
I’ve pulled `master` into #90 and have added some adjustments - let’s land this thing!

cc @pendowski thank you so much for this feature!!

### CODE REVIEW

**Comment by pendowski** [comment]
Too bad you removed the `viewController` argument from the callback. 
From what I can see in the app where Tweaks are used there's no callback tweak that doesn't use it. 

Sometimes it's used to push another view (to see logs) in other instances it's used to display `UIAlertController` for confirmation or to make a choice (which recorded state should be reset). 
There are other ways to achieve that but usually, it involves playing with some global object or moving tweaks just to have `window`/`viewController` reference.

In other words - it is very useful to have that `viewController` reference (sender wasn't that useful - maybe would be with some popup UI) and I'm not sure what's the harm of exposing generic `UIViewController` reference

**Comment by bryanjclark** [comment]
@pendowski I agree, it’s a useful idea! I wrote a bit about why I removed it in the commit message:

> (While I recognize that there are some neat abilities lost by changing the parameters for a `TweakActionClosure` — we can build those abilities without exposing the internal `UIViewControllers` of SwiftTweaks to outside developers! I really like the feature idea, though - embedding custom debug menu interfaces would be super neat! I'd like to do this in a different way, though, rather than adding extra bloat to `TweakAction` to do it.)

What I’m trying to say here ^ is that:
 - SwiftTweaks _intentionally_ keeps its internals internal-to-SwiftTweaks, so that there are fewer breaking changes for adopters, and any future-changes to SwiftTweaks’ UI don’t have to consider whether-or-not they’re breaking existing usage.
 - The “embed a view controller in the SwiftTweaks menu” is a really neat idea — but it’s a very _different_ idea from the simpler one expressed in #19, and warrants a different development approach, rather than folding it in to the idea of “run a `() -> Void` closure”

(I agree with you that hosting a custom view controller is a neat idea (like an analytics view, or a browser pointed at an internal web page, or any number of other handy debug interface bits) but it would belong in a different place!)

File: SwiftTweaks/TweakAction.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+public class TweakAction {
+	public enum Error: Swift.Error {
+		case wrongIdentifier
+	}
+	
+	public typealias ClosureIdentifier = UInt
+	
+	private var lastToken: ClosureIdentifier = 0
+	private var closures: [ClosureIdentifier: TweakActionClosure] = [:]
```
Comment (nixterrimus) [inline SwiftTweaks/TweakAction.swift]: This should maybe hold a weak reference to the callbacks

**Comment by pendowski** [comment]
> SwiftTweaks intentionally keeps its internals internal-to-SwiftTweaks, so that there are fewer breaking changes for adopters, and any future-changes to SwiftTweaks’ UI don’t have to consider whether-or-not they’re breaking existing usage.

As much as I agree with the breaking changes I don't think reference to `UIViewController` will cause that in foreseeable future. You nor Apple will rewrite `UIViewController`s, `navigationController` (which we use to push custom views) is an optional - I don't think these are the places where compatibility will cause an issue. 

And even if for some reason it would change - it's better to be informed that internal UI-side of SwiftTweaks changed (by getting different object in the callback or no object at all) than to realise that a hack that you did to be able to show something on top of Tweaks stopped working. Which will likely happen with callbacks - as I said we have multiple callback tweaks in the working app and when I checked today - all of them make use of that `viewController` reference, none of them use the `sender` reference - proving in real-life application that one was a good and useful idea and the other one wasn't.

Keep in mind that Tweaks should be used on a device by a much more "normal" user. A simple thing as showing confirmation (but not in the Xcode console) that the callback did run is much better UX than nothing at all - that's the simplest use of that `viewController` reference.

**Comment by bryanjclark** [comment]
> Keep in mind that Tweaks should be used on a device by a much more "normal" user. A simple thing as showing confirmation (but not in the Xcode console) that the callback did run is much better UX than nothing at all

That’s a great point! I think the SwiftTweaks UI could do with an animation or some other “more apparent” confirmation that a closure has been run.

File: SwiftTweaks/TweakAction.swift (hunk 1 above)
Comment (bryanjclark) [reply SwiftTweaks/TweakAction.swift]: Ah, I can't add a `weak` reference here because the compiler only allows weak references for `class` types!
